### ISSUE TITLE
conda-build 3.5.0 cannot detect binary files on Windows

### ISSUE THREAD

**Issue by peterjc123** [conda/conda-build#2698]
### Actual Behavior

The build phase ended with the message `No files or script found for output xxxx`, and the test phase failed with the message `No module named xxxx`. The package generated contains only the recipe data. However, the build was successful and no error was triggered.

### Expected Behavior

This package should detect and contain binary files. When I downgrade conda-build to 3.0.1, everything works fine.

### Steps to Reproduce

1. Use the new compiler config listed in #2541.
2. conda skeleton some python packages that can be built using MSVC 2017
3. Copy conda-build-config.yaml from step 1 into the package dir, and include `compiler('c')` as a requirement of build.
4. Start building use sth like `conda build .`

##### `Output of conda info`

```
(base) C:\Users\Administrator>conda info

     active environment : base
    active env location : c:\Miniconda2
            shell level : 1
       user config file : C:\Users\Administrator\.condarc
 populated config files : C:\Users\Administrator\.condarc
          conda version : 4.4.10
    conda-build version : 0+unknown
         python version : 2.7.14.final.0
       base environment : c:\Miniconda2  (writable)
           channel URLs : https://repo.continuum.io/pkgs/main/win-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
          package cache : c:\Miniconda2\pkgs
                          C:\Users\Administrator\AppData\Local\conda\conda\pkgs
       envs directories : c:\Miniconda2\envs
                          C:\Users\Administrator\AppData\Local\conda\conda\envs
                          C:\Users\Administrator\.conda\envs
               platform : win-64
             user-agent : conda/4.4.10 requests/2.18.4 CPython/2.7.14 Windows/2012ServerR2 Windows/6.3.9600
          administrator : True
             netrc file : None
           offline mode : False
```

**Comment by msarahan** [comment]
which recipe, exactly?  I have built cytoolz for python 3.6 fine using vs2017_win-64.

I suspect a recipe issue more than a conda-build issue.  In particular, conda-build 3.0.1 predates the split of the build and host prefixes.  It merges them if the target_platform is the same as the native platform.  This changed in conda-build 3.1.4, and we also bumped the minor version at 3.2.0 to reflect it  https://github.com/conda/conda-build/releases/tag/3.2.0.  If your recipe has python in the build requirements, not the host requirements, your package will end up empty - the host prefix is the one that gets bundled.
